### PR TITLE
Completed the translation of the first part of Chapter 7 into Russian.

### PR DESCRIPTION
**Pull request by blademoon**
#75

This is the first part of the translation of chapter 7 of the course. Since this chapter is very massive, it was decided to split the translation into two parts.

@MKhalusova I'm waiting for your review Maria)

### CODE REVIEW

File: chapters/ru/chapter7/4.mdx [hunk 1]
```diff
@@ -0,0 +1,1002 @@
...
+
+optimizer, schedule = create_optimizer(
+    init_lr=5e-5,
+    num_warmup_steps=0,
+    num_train_steps=num_train_steps,
+    weight_decay_rate=0.01,
+)
+model.compile(optimizer=optimizer)
+
+# Train in mixed-precision float16
```
Comment (MKhalusova) [inline chapters/ru/chapter7/4.mdx:629]: Please translate the comment

File: chapters/ru/chapter7/5.mdx [hunk 2]
```diff
@@ -0,0 +1,1073 @@
...
+```python
+inputs = tokenizer("I loved reading the Hunger Games!")
+inputs
+```
+
+```python out
+{'input_ids': [336, 259, 28387, 11807, 287, 62893, 295, 12507, 1], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1]}
+```
+
+Здесь мы видим знакомые нам `input_ids` и `attention_mask`, с которыми мы столкнулись в наших первых экспериментах по дообучению еще в [Главе 3] (/course/chapter3). Давайте декодируем эти входные идентификаторы с помощью функции токенизатора `convert_ids_to_tokens()`, чтобы понять, с каким токенизатором мы имеем дело:
```
Comment (MKhalusova) [inline chapters/ru/chapter7/5.mdx:268]:
```suggestion
Здесь мы видим знакомые нам `input_ids` и `attention_mask`, с которыми мы столкнулись в наших первых экспериментах по дообучению еще в [Главе 3](/course/chapter3). Давайте декодируем эти входные идентификаторы с помощью функции токенизатора `convert_ids_to_tokens()`, чтобы понять, с каким токенизатором мы имеем дело:
```

File: chapters/ru/chapter7/5.mdx [hunk 3]
```diff
@@ -0,0 +1,1073 @@
...
+
+```python
+tokenizer.convert_ids_to_tokens(inputs.input_ids)
+```
+
+```python out
+['▁I', '▁', 'loved', '▁reading', '▁the', '▁Hung', 'er', '▁Games', '</s>']
+```
+
+Специальный символ Юникода `▁` и токен конца последовательности `</s>` указывают на то, что мы имеем дело с токенизатором SentencePiece, который основан на алгоритме сегментации Unigram, рассмотренном в [Главе 6] (/course/chapter6). Unigram особенно полезен для многоязычных корпусов, поскольку он позволяет SentencePiece не зависеть от ударений, пунктуации и того факта, что во многих языках, например в японском, нет пробельных символов.
```
Comment (MKhalusova) [inline chapters/ru/chapter7/5.mdx:278]:
```suggestion
Специальный символ Юникода `▁` и токен конца последовательности `</s>` указывают на то, что мы имеем дело с токенизатором SentencePiece, который основан на алгоритме сегментации Unigram, рассмотренном в [Главе 6](/course/chapter6). Unigram особенно полезен для многоязычных корпусов, поскольку он позволяет SentencePiece не зависеть от ударений, пунктуации и того факта, что во многих языках, например в японском, нет пробельных символов.
```

File: chapters/ru/chapter7/5.mdx [hunk 4]
```diff
@@ -0,0 +1,1073 @@
...
+{#if fw === 'pt'}
+
+Для вычисления оценки ROUGE в процессе обучения нам понадобится генерировать резюме. К счастью, 🤗 Transformers предоставляет специальные классы `Seq2SeqTrainingArguments` и `Seq2SeqTrainer`, которые могут сделать это за нас автоматически! Чтобы увидеть, как это работает, давайте сначала определим гиперпараметры и другие аргументы для наших экспериментов:
+
+```python
+from transformers import Seq2SeqTrainingArguments
+
+batch_size = 8
+num_train_epochs = 8
+# Show the training loss with every epoch
```
Comment (MKhalusova) [inline chapters/ru/chapter7/5.mdx:511]: Please translate the comment

File: chapters/ru/chapter7/5.mdx [hunk 5]
```diff
@@ -0,0 +1,1073 @@
...
+
+Давайте посмотрим, что выдает этот коллатор, когда ему передается небольшой батч примеров. Во-первых, нам нужно удалить столбцы со строками, потому что коллатор не будет знать, как вставлять эти элементы:
+
+```python
+tokenized_datasets = tokenized_datasets.remove_columns(
+    books_dataset["train"].column_names
+)
+```
+
+Давайте посмотрим, что выдает этот коллатор, когда ему передается небольшой батч примеров. Во-первых, нам нужно удалить столбцы со строками, потому что коллатор не будет знать, как вставлять эти элементы:
```
Comment (MKhalusova) [inline chapters/ru/chapter7/5.mdx:592]: This is a copy of the previous paragraph. What was supposed to be here is: "Since the collator expects a list of dicts, where each dict represents a single example in the dataset, we also need to wrangle the data into the expected format before passing it to the data collator:"

File: chapters/ru/chapter7/5.mdx [hunk 6]
```diff
@@ -0,0 +1,1073 @@
...
+```
+
+
+{/if}
+
+{#if fw === 'pt'}
+
+## Дообучение mT5 с 🤗 Accelerate[[fine-tuning-mt5-with-accelerate]]
+
+Дообучение нашей модели с помощью 🤗 Accelerate очень похоже на пример с классификацией текста, который мы рассматривали в [Главе 3] (/course/chapter3). Основные отличия заключаются в необходимости явной генерации резюме во время обучения и определении способа вычисления оценок ROUGE (напомним, что `Seq2SeqTrainer` позаботился о генерации за нас). Давайте посмотрим, как мы можем реализовать эти два требования в 🤗 Accelerate!
```
Comment (MKhalusova) [inline chapters/ru/chapter7/5.mdx:795]:
```suggestion
Дообучение нашей модели с помощью 🤗 Accelerate очень похоже на пример с классификацией текста, который мы рассматривали в [Главе 3](/course/chapter3). Основные отличия заключаются в необходимости явной генерации резюме во время обучения и определении способа вычисления оценок ROUGE (напомним, что `Seq2SeqTrainer` позаботился о генерации за нас). Давайте посмотрим, как мы можем реализовать эти два требования в 🤗 Accelerate!
```

**Review by MKhalusova — COMMENTED** [review]
Great work! Left a few nits.

File: chapters/ru/chapter7/1.mdx [hunk 7]
```diff
@@ -0,0 +1,38 @@
+<FrameworkSwitchCourse {fw} />
+
+# Введение[[introduction]]
+
+<CourseFloatingBanner
+    chapter={7}
+    classNames="absolute z-10 right-0 top-0"
+/>
+
+В [Главе 3] (/course/chapter3) вы узнали, как дообучить модель для классификации текстов. В этой главе мы рассмотрим следующие общие задачи NLP:
```
[the remark this answers is not in the document]
Comment (blademoon) [reply chapters/ru/chapter7/1.mdx:10]: Yeah, I didn't notice.

File: chapters/ru/chapter7/4.mdx [hunk 8]
```diff
@@ -0,0 +1,1002 @@
...
+
+- **Перенос стиля (Style transfer)**: Создание модели, которая *переводит* тексты, написанные в определенном стиле, в другой (например, из формального в повседневный или из шекспировского английского в современный).
+- **Генеративные ответы на вопросы (Generative question answering)**: Создание модели, которая генерирует ответы на вопросы, учитывая контекст
+
+<Youtube id="1JvfrvZgi6c"/>
+
+Если у вас есть достаточно большой корпус текстов на двух (или более) языках, вы можете обучить новую модель перевода с нуля, как мы это сделаем в разделе по [казуальному языковому моделированию (causal language modeling)] (/course/chapter7/6). Однако быстрее будет дообучить существующую модель перевода, будь то многоязычная модель типа mT5 или mBART, которую нужно дообучить для конкретной пары языков, или даже модель, специализированная для перевода с одного языка на другой, которую нужно дообучить для конкретного корпуса.
```
[the remark this answers is not in the document]
Comment (blademoon) [reply chapters/ru/chapter7/4.mdx:32]: Extra space)

File: chapters/ru/chapter7/5.mdx (hunk 4 above)
Comment (blademoon) [reply chapters/ru/chapter7/5.mdx]: Problem solved in 95bda7c413f0b98fa47e219a587706b677299cd8

File: chapters/ru/chapter7/2.mdx [hunk 9]
```diff
@@ -0,0 +1,981 @@
...
+tf_eval_dataset = tokenized_datasets["validation"].to_tf_dataset(
+    columns=["attention_mask", "input_ids", "labels", "token_type_ids"],
+    collate_fn=data_collator,
+    shuffle=False,
+    batch_size=16,
+)
+```
+
+
+ Next stop: the model itself.
```
[the remark this answers is not in the document]
Comment (blademoon) [reply chapters/ru/chapter7/2.mdx]: Resolved in 913a9b1efb91988fc374f6c525f4b6b5dc1b1ef5

File: chapters/ru/chapter7/3.mdx [hunk 10]
```diff
@@ -0,0 +1,1044 @@
...
+<div class="flex justify-center">
+<img src="https://huggingface.co/datasets/huggingface-course/documentation-images/resolve/main/en/chapter7/mlm-models.png" alt="Hub models." width="80%"/>
+</div>
+
+Хотя модели семейства BERT и RoBERTa являются наиболее загружаемыми, мы будем использовать модель под названием [DistilBERT](https://huggingface.co/distilbert-base-uncased) 
+которая может быть обучена гораздо быстрее и практически без потерь в качестве. Эта модель была обучена с помощью специальной техники, называемой [_дистилляцией знаний (knowledge distillation)_](https://en.wikipedia.org/wiki/Knowledge_distillation), когда большая "модель-учитель", такая как BERT, используется для обучения "модели-ученика", имеющей гораздо меньше параметров. Объяснение деталей дистилляции знаний завело бы нас слишком далеко в этом разделе, но если вам интересно, вы можете прочитать об этом в [_Natural Language Processing with Transformers_](https://www.oreilly.com/library/view/natural-language-processing/9781098136789/) (в просторечии известном как учебник по Трансформерам).
+
+{#if fw === 'pt'}
+
+Let's go ahead and download DistilBERT using the `AutoModelForMaskedLM` class:
```
[the remark this answers is not in the document]
Comment (blademoon) [reply chapters/ru/chapter7/3.mdx]: Solved in 5f5d4aaf7cc35aebfc1dcb7d80e21cfa1301113e

File: chapters/ru/chapter7/3.mdx [hunk 11]
```diff
@@ -0,0 +1,1044 @@
...
+'>>> Label: 0'
+
+'>>> Review: I saw this movie at the theaters when I was about 6 or 7 years old. I loved it then, and have recently come to own a VHS version. <br /><br />My 4 and 6 year old children love this movie and have been asking again and again to watch it. <br /><br />I have enjoyed watching it again too. Though I have to admit it is not as good on a little TV.<br /><br />I do not have older children so I do not know what they would think of it. <br /><br />The songs are very cute. My daughter keeps singing them over and over.<br /><br />Hope this helps.'
+'>>> Label: 1'
+```
+
+Да, это точно рецензии на фильмы, и если вы достаточно взрослый, то можете даже понять комментарий в последней рецензии о том, что у вас есть VHS-версия 😜! Хотя нам не понадобятся эти метки для языкового моделирования, мы уже видим, что `0` обозначает отрицательный отзыв, а `1` - положительный.
```
[the remark this answers is not in the document]
Comment (blademoon) [reply chapters/ru/chapter7/3.mdx]: Согласен. Внес соответствующие исправления в 30ca4467842d7f323ed1b038a24618234aabd4cf .

File: chapters/ru/chapter7/3.mdx [hunk 12]
```diff
@@ -0,0 +1,1044 @@
...
+
+## Предварительная обработка данных[[preprocessing-the-data]]
+
+<Youtube id="8PmhEIXhBvI"/>
+
+Как для авторегрессивного, так и для масочного моделирования языка общим шагом предварительной обработки является объединение всех примеров, а затем разбиение всего корпуса на части одинакового размера. Это сильно отличается от нашего обычного подхода, когда мы просто проводим токенизацию отдельных примеров. Зачем конкатенируем все вместе? Причина в том, что отдельные примеры могут быть обрезаны, если они слишком длинные, и это приведет к потере информации, которая может быть полезна для задачи языкового моделирования!
+
+Итак, для начала мы проведем обычную токенизацию нашего корпуса, но _без_ задания параметра `truncation=True` в нашем токенизаторе. Мы также возьмем идентификаторы слов, если они доступны ((а они будут доступны, если мы используем быстрый токенизатор, как описано в [Главе 6](/course/chapter6/3)), поскольку они понадобятся нам позже для маскирования целых слов. Мы обернем это в простую функцию, а пока удалим столбцы `text` и `label`, поскольку они нам больше не нужны:
```
[the remark this answers is not in the document]
Comment (blademoon) [reply chapters/ru/chapter7/3.mdx:254]: An unnecessary parenthesis.

File: chapters/ru/chapter7/3.mdx [hunk 13]
```diff
@@ -0,0 +1,1044 @@
...
+num_train_steps = len(tf_train_dataset)
+optimizer, schedule = create_optimizer(
+    init_lr=2e-5,
+    num_warmup_steps=1_000,
+    num_train_steps=num_train_steps,
+    weight_decay_rate=0.01,
+)
+model.compile(optimizer=optimizer)
+
+# Train in mixed-precision float16
```
Comment (blademoon) [inline chapters/ru/chapter7/3.mdx]: Done in 1a195753968baef08a2bb1c42f6bb13f1dd696a5

[the remark this answers is not in the document]
Comment (blademoon) [reply chapters/ru/chapter7/3.mdx]: Done in 1a195753968baef08a2bb1c42f6bb13f1dd696a5

File: chapters/ru/chapter7/4.mdx (hunk 1 above)
Comment (blademoon) [reply chapters/ru/chapter7/4.mdx]: Done in 5eb4f4be1320f29454758cc7def43e1bb2e4f1a0

File: chapters/ru/chapter7/5.mdx (hunk 5 above)
Comment (blademoon) [reply chapters/ru/chapter7/5.mdx]: Done in e7c59a56dd985228fe3670799d9492e986fae319

**Review by blademoon — COMMENTED** [review]
@MKhalusova All done. You can check)))

**Review by MKhalusova — APPROVED** [review]
LGTM!
